spec/reek/smells/nested_iterators_spec.rb

```diff
@@ -127,6 +107,72 @@ def super_call_with_block
     expect(src).to reek_of(:NestedIterators)
   end
 
+  describe 'examine_context / warnings' do
+    let(:detector) { build(:smell_detector, smell_type: :NestedIterators) }
+
+    it 'reports correctly' do
+      source = <<-EOS
+        def foo
+          bar { baz { } }
+        end
+      EOS
+      warnings = detector.examine_context(create_method_context(source))
+      expect(warnings.size).to eq(1)
+      warning = warnings.first
+
+      expect(warning.smell_category).to eq(Reek::Smells::NestedIterators.smell_category)
+      expect(warning.smell_type).to eq(Reek::Smells::NestedIterators.smell_type)
+      expect(warning.parameters[:name]).to eq('foo')
+      expect(warning.lines).to eq([2])
+    end
+
+    it 'should report nested iterators only once per method' do
+      source = <<-EOS
```

Comment:
And here are those two specs from above again, this time fullfilling the promise.